internal/controllers/module_reconciler.go

```diff
@@ -103,32 +100,14 @@ func (mr *ModuleReconciler) Reconcile(ctx context.Context, mod *kmmv1beta1.Modul
 		return ctrl.Result{}, fmt.Errorf("failed to get list of nodes by selector: %v", err)
 	}
 
-	currentNMCs, err := mr.reconHelper.getNMCsByModuleSet(ctx, mod)
```

Comment:
i would leave getNMCsByModuleSet and prepareSchedulingData in the reconciler.  Reconciler should be clear, but not empty. In addition gathering all those functions in one function will make unit-testing harder

Reply:
> i would leave getNMCsByModuleSet and prepareSchedulingData in the reconciler

But those 2 function are only used for NMC handling and not for anything else.

> In addition gathering all those functions in one function will make unit-testing harder

On the contrary, they make them easier because each function is smaller, and all sub-fucntions that were previously used are still used so no complexity is added (it is just breaking the controller logic into multiple function but all the rest stays the same.

Reply:
once you unite those functions in one function, your unit-test needs to test all those functions together, since they are in the same interface. there is nothing wrong in Reconciler containing code, as long as it is clear. Having those functions in the Reconciler allows others to look at the reconciler function and understand the flow, without the need to descend into all those helper functions